### PR TITLE
Use docopt for argument parsing

### PR DESCRIPTION
**Pull request by fitzgen**
This commit switches bindgen over to using the docopt crate for argument parsing
instead of manual argument parsing. This required two notable changes in the
arguments and flags style:

1. All flags of the form `-foo` are now of the form `--foo`.

2. We can no longer pass unknown flags straight through to clang. Instead, the
user appends `--` after the bindgen flags and input header, after which point
any more flags and arguments get collected and passed to clang. This required
changes to the test runner and the `// bindgen-flags` comments.

r? @emilio

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,38 +1,49 @@
...
+version = "0.21.1"
+
+[dependencies.clippy]
+optional = true
+version = "*"
+
+[dependencies.quasi]
+features = ["with-syntex"]
+version = "0.15"
+
```
Comment (fitzgen) [inline Cargo.toml]: Using `cargo add` really goes to town with reformatting toml files... I can revert this and add the dependency manually, if you'd prefer.

**Comment by emilio** [comment]
Wohoo, thanks for doing this! :-)

I'll merge it, though now I'll have to rebase stuff and change scripts :P 

@bors-servo: r+

**Comment by Manishearth** [comment]
This needs a corresponding PR to the servo side (regen.py), btw

**Comment by emilio** [comment]
Was the first thing I did a few hours ago: https://github.com/servo/servo/pull/13079

Feel free to review it :P

**Comment by fitzgen** [comment]
Thanks for merging :)

Will need to update `servo/rust-mozjs/etc/bindings.sh` too -- I can do that after the smup lands.

**Pull request merged**